packages/kit/src/exports/public.d.ts

```diff
@@ -742,6 +742,43 @@ export type ClientInit = () => MaybePromise<void>;
  */
 export type Reroute = (event: { url: URL }) => void | string;
 
+/**
+ * The [`transport`](https://svelte.dev/docs/kit/hooks#Universal-hooks-transport) hook allows you to transport custom types across the server/client boundary.
+ *
+ * Each transporter has a pair of `encode` and `decode` functions. On the server, `encode` determines whether a value is an instance of the custom type and, if so, returns a non-falsy encoding of the value which can be an object or an array (or `false` otherwise).
+ *
+ * In the browser, `decode` turns the encoding back into an instance of the custom type.
+ *
+ * ```ts
+ * import type { Transport } from '@sveltejs/kit';
+ *
+ * declare class MyCustomType {
+ * 	data: any
+ * }
+ *
+ * // hooks.js
+ * export const transport: Transport = {
+ * 	MyCustomType: {
+ * 		encode: (value) => value instanceof MyCustomType && [value.data],
+ * 		decode: ([data]) => new MyCustomType(data)
+ * 	}
+ * };
+ * ```
+ * @since 2.11.0
+ */
+export type Transport = Record<string, Transporter>;
```

Comment:
How much use will people really make of this specific type, since it's not helping much with type safety? Should we just omit it?

Reply:
I still think we should at least offer the `defineTransport` function for type safety. We can call it something else if we don't like it but it would be nice

Reply:
Yes and no - we can add `satisfies Transport` for the top level, but we can't do much for the individual entries